### ISSUE TITLE
about:create_train_test_set.py

### ISSUE THREAD

**Issue by Apollo0801** [munhouiani/Deep-Packet#17]
Why set _test_size= 0.2_ in _create_train_test_set.py_, but the resulting data set (training set: Test Set) is not (8:2). Moreover, the sample size of the test set is much higher than that of the training set.

**Comment by munhouiani** [comment]
Because of undersampling.

We first split the entire set into train and test at [lines 54 to 65](https://github.com/munhouiani/Deep-Packet/blob/e2d5bf20e97e96f18767dfad2ec467d030dd3c38/create_train_test_set.py#L54). The ratio of train and test should be around 8:2.

At [line 68](https://github.com/munhouiani/Deep-Packet/blob/e2d5bf20e97e96f18767dfad2ec467d030dd3c38/create_train_test_set.py#L68), if `under_sampling_train` is set to `True`, we balance the train set by undersampling. 

That is the reason why the final train set is smaller than test set.

**Comment by Apollo0801** [comment]
> Because of undersampling.
> 
> We first split the entire set into train and test at [lines 54 to 65](https://github.com/munhouiani/Deep-Packet/blob/e2d5bf20e97e96f18767dfad2ec467d030dd3c38/create_train_test_set.py#L54). The ratio of train and test should be around 8:2.
> 
> At [line 68](https://github.com/munhouiani/Deep-Packet/blob/e2d5bf20e97e96f18767dfad2ec467d030dd3c38/create_train_test_set.py#L68), if `under_sampling_train` is set to `True`, we balance the train set by undersampling.
> 
> That is the reason why the final train set is smaller than test set.

Thank you very much for your answer.

**Comment by dimitrov89** [comment]
If we have almost the same amount of packets for every label, can we skip undersampling and have around 8:2 training:testing?
The count in train set is always the same for every label.
How to get the same (or similar) count for every label in the test set?
